src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -462,7 +462,7 @@ class MySimulation : public SimulationHelper<simDim>
         auto fieldE = dc.get< FieldE >( FieldE::getName(), true );
         auto fieldB = dc.get< FieldB >( FieldB::getName(), true );
 
-        if( step != 0 )
+        if( this->restartRequested )
         {
             namespace nvfct = PMacc::nvidia::functors;
 
```

Comment:
my first comment was just about that the PR description does not state that it fixes the time offset (only the title).

my second comment is, that the `+1u` needs to be added to line 484 to fix `dev`